### PR TITLE
perforce: Test sample 'protects -a' output

### PR DESCRIPTION
**Pull request by ryanslade**
Refactored the code to make it easer to test the ouput of a perforce
command and specifically tested sample `protects -a' output.

### CODE REVIEW

**Review by asdine — APPROVED** [review]
LGTM

File: internal/authz/perforce/perforce.go [hunk 1]
```diff
@@ -194,14 +204,19 @@ func (p *Provider) FetchUserPerms(ctx context.Context, account *extsvc.Account)
 	for scanner.Scan() {
 		line := scanner.Text()
 
+		// Skip comments
+		if strings.HasPrefix(line, "#") {
```
Comment (unknwon) [inline internal/authz/perforce/perforce.go:208]: Technically, [comments in the Perforce protections table start with `##` not single `#`](https://www.perforce.com/manuals/p4sag/Content/P4SAG/security.protect.comments.html?TocPath=Securing%20the%20server%7CAuthorizing%20access%7C_____4).

File: internal/authz/perforce/perforce.go [hunk 2]
```diff
@@ -350,25 +365,58 @@ func (p *Provider) FetchRepoPerms(ctx context.Context, repo *extsvc.Repository)
...
+// return a map of users
+func (p *Provider) scanAllUsers(ctx context.Context, rc io.ReadCloser) (map[string]struct{}, error) {
 	users := make(map[string]struct{})
 	scanner := bufio.NewScanner(rc)
 	for scanner.Scan() {
 		line := scanner.Text()
 
-		// Trim comments
+		// Skip comments
+		if strings.HasPrefix(line, "#") {
```
Comment (unknwon) [inline internal/authz/perforce/perforce.go:405]: ditto

**Review by unknwon — APPROVED** [review]
Thanks! Left few comments.

File: internal/authz/perforce/perforce.go (hunk 1 above)
Comment (ryanslade) [reply internal/authz/perforce/perforce.go:208]: That's weird... the sample we received has single hashes

Comment (unknwon) [reply internal/authz/perforce/perforce.go]: Hmm... that's strange, we probably also want to confirm that with the customer 🤔

**Pull request merged**